code/modules/research/rdconsole.dm

```diff
@@ -166,61 +154,65 @@ won't update every console in existence) but it's more of a hassle to do. Also,
+			var/list/disk_designs = disk.find_files_by_type(/datum/computer_file/binary/design)
 
-	else if(href_list["eject_tech"] && t_disk) //Eject the technology disk.
-		t_disk.forceMove(get_turf(src))
-		t_disk = null
-		screen = 1.0
+			for(var/f in disk_designs)
+				var/datum/computer_file/binary/design/design_file = f
 
-	else if(href_list["copy_tech"] && t_disk) //Copys some technology data from the research holder to the disk.
-		for(var/datum/tech/T in files.known_tech)
-			if(href_list["copy_tech_ID"] == T.id)
-				t_disk.stored = T
-				break
-		screen = 1.2
+				if(design_file.copy_protected)
+					continue
+				if(!(design_file.design in files.possible_designs))
+					continue
 
-	else if(href_list["updt_design"] && d_disk && length(d_disk.recipes)) //Updates the research holder with design data from the design disk.
-		screen = 1.4
+				files.AddDesign2Known(design_file.design)
 
-		for(var/d in d_disk.recipes)
-			var/datum/design/design = files.possible_design_ids[d]
-			if(!design)
-				continue
 
-			files.AddDesign2Known(design)
+			var/list/disk_technologies = disk.find_files_by_type(/datum/computer_file/binary/tech)
 
-		updateUsrDialog()
-		griefProtection() //Update centcomm too
+			for(var/f in disk_technologies)
+				var/datum/computer_file/binary/tech/technology_file = f
+				files.AddTech2Known(technology_file.tech)
+
+			updateUsrDialog()
+			griefProtection() //Update centcomm too
+
+		else if(href_list["copy_tech"]) //Copys some technology data from the research holder to the disk.
+			var/datum/tech/tech
+			for(var/datum/tech/T in files.known_tech)
+				if(href_list["copy_tech"] == T.id)
+					tech = T
+					break
+
+			if(tech)
+				var/datum/computer_file/binary/tech/tech_file = new
+				tech_file.set_tech(tech.Copy())
+				disk.store_file(tech_file)
 
-	else if(href_list["eject_design"] && d_disk) //Eject the design disk.
-		d_disk.forceMove(get_turf(src))
-		d_disk = null
-		screen = 1.0
 
-	else if(href_list["copy_design"] && d_disk) //Copy design data from the research holder to the design disk.
-		var/datum/design/design = files.possible_design_ids[href_list["copy_design_ID"]]
+		else if(href_list["copy_design"]) //Copy design data from the research holder to the design disk.
+			var/datum/design/design = files.possible_design_ids[href_list["copy_design"]]
 
-		if(length(d_disk.recipes) < 10)
-			if(design in files.known_designs)
-				d_disk.recipes |= design.id
-				d_disk.category = "Research Designs"
+			if(design)
+				disk.store_file(design.file.clone())
 
-		screen = 1.4
+
+	if(href_list["menu"]) //Switches menu screens. Converts a sent text string into a number. Saves a LOT of code.
+		var/temp_screen = text2num(href_list["menu"])
+		if(temp_screen <= 1.1 || (3 <= temp_screen && 4.9 >= temp_screen) || allowed(usr) || emagged) //Unless you are making something, you need access.
+			screen = temp_screen
+		else
+			usr << "Unauthorized Access."
```

Comment:
`to_chat(usr, "Unauthorized Access.")`?

Reply:
Strange, I don't remember touching this piece of code. Guess I touched everything around it or something.